### PR TITLE
[WIP] ES6 + Linting

### PR DESCRIPTION
**Pull request by vramana**

### CODE REVIEW

File: public/js/histograms.js [hunk 1]
```diff
@@ -77,82 +71,79 @@ module.exports = function createHistogram(counts, win_counts, label) {
...
                         t = Number(t) / scalef;
-                        //these labels are times and need to be formatted
+                        // these labels are times and need to be formatted
                         var times = {
-                            "duration": 1,
-                            "first_blood_time": 1
+                            duration: 1,
+                            first_blood_time: 1
                         };
                         if (times[label]) {
```
Comment (vramana) [inline public/js/histograms.js]: @howardc93 @albertcui Isn't this reduntant code here? `times` object is has no key label clearly from the declaration.

Comment (howardc93) [inline public/js/histograms.js]: label is a variable.  So we may have that key.

Comment (vramana) [inline public/js/histograms.js]: Sorry my bad

**Comment by howardc93** [comment]
How thoroughly is this tested?

**Comment by vramana** [comment]
Hey, I haven't tested this. My machine is taking too long to even build. So, I didn't test this yesterday I will today. In the meantime Can you build this locally once and report any problems? I don't expect anything to break as such.

**Comment by vramana** [comment]
I have tested this branch by running `npm start web` everything seems to render fine except I don't have redis database. May be you can check that part.

**Comment by howardc93** [comment]
@albertcui do you want to weigh in on ES6?  Eventually we'll be able to avoid the need to use babel, right?

Does the eslint slow the build significantly? (How long does it take to run?)

**Comment by vramana** [comment]
@howardc93 It doesn't slow the build at all. May be it takes around 5-10 secs to do the linting.

**Comment by howardchung** [comment]
Playing a bit with React now so I've added babel to the build step and can now use ES6.  Unfortunately the code has changed a LOT since the last commit.

I'm still hoping eventually we won't need babel anymore.

**Comment by howardchung** [comment]
Please see https://github.com/yasp-dota/ui if you'd like to help contribute to the new UI.  We're using ES6 for that.

**Pull request closed without merging**